doc/source/generate_module_rst.py

```diff
@@ -134,7 +135,7 @@ def get_component_rst_string(module: ModuleType, component: Callable, level: int
     object_name = f"{module.__name__}.{component.__name__}"
 
     rst_documentation = ""
-    level_underline = RST_LEVEL_SYMBOLS[level] * 6
+    level_underline = RST_LEVEL_SYMBOLS[level] * len(object_name)
```

Comment:
This (and the related changes below) fixes a bunch of warnings about wrong length of underlines when running sphinx.